components/composable/notebook.js

```diff
@@ -0,0 +1,399 @@
+// @flow
+import React from "react";
+
+import SyntaxHighlighter from "react-syntax-highlighter";
+import ReactMarkdown from "react-markdown";
+import { idea } from "react-syntax-highlighter/dist/styles";
+
+export const Passthrough = (props: { children?: React$Node }) =>
+  props.children || null;
+
+export const Output = () => <pre>I am output</pre>;
+
+type OutputsProps = {
+  children: React$Node,
+  className?: string
```

Comment:
Should the output prompt have a prop that indicates it's hidden/non-hidden status as well?

That way in the render we can do something like:

```
if (this.props.children && this.props.visible) { ... } else { return null };
```

Not sure if this is the right kind of state to have in the prop though.

Reply:
I don't have a good heuristic here, I guess we'd have to compare whether we want it to be on the outside or not (also note that I used `hidden` as the prop since I'd want the default to be visible with an explicit property when not):

```
<Cell>
  <Input />
  <Outputs hidden />
</Cell>
```

vs.

```
<Cell>
  <Input />
  { visible ? <Outputs /> : null }
</Cell>
```

People can always do the second one whether or not we have the `hidden` interface. I have to admit I prefer the first, even though I'm not sure what benefit it would have inside the component

Reply:
> also note that I used hidden as the prop since I'd want the default to be visible with an explicit property when not

That makes sense!

I agree with you. IMO, having it within the component feels like a more intuitive option for the end user.
1. It reduces the amount of code that I have to write.
2. It's much easier to see that `hidden` is associated with the `Output`.